var/spack/repos/builtin/packages/daos/package.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Daos(SConsPackage):
+    """The Distributed Asynchronous Object Storage (DAOS) is an open-source
+    software-defined object store designed from the ground up for massively
+    distributed Non Volatile Memory (NVM)."""
+
+    homepage = "https://github.com/daos-stack/daos"
+    git = "https://github.com/daos-stack/daos.git"
+    maintainers("hyoklee")
+
+    version("master", branch="master", submodules=True)
+    version("2.2.0", tag="v2.2.0", submodules=True)
+    variant(
+        "debug", default=False, description="Enable debugging info and strict compile warnings"
+    )
+
+    depends_on("argobots@1.1:")
+    depends_on("boost@develop+python", type="build")
+    depends_on("cmocka", type="build")
+    depends_on("dpdk@22.11:")
```

Comment:
I fear it's a little risky to include dpdk separately (maybe it's fine but double check) because spdk already includes dpdk as a submodule so you'd have to make sure it's the same versions I think.

Reply:
if you use the variant I mentioned in the other spdk PR, you won't need that dependency here anymore and instead you'd just do `depends_on("spdk@22.01.1:+shared+rdma+dpdk")`